src/Writer/MSSQL.php

```diff
@@ -135,7 +135,8 @@ public function write(CsvFile $csv, array $table): void
             if (!empty($col['nullable'])) {
                 $srcColName = sprintf("NULLIF(%s, '')", $colName);
             }
-            $column = sprintf('TRY_CAST(%s AS %s%s) as %s', $srcColName, $type, $size, $colName);
+            $column = $this->bcpCast($srcColName, $type, $size, $colName, $this->dbParams['tdsVersion']);
```

Comment:
I don't understand where does this tdsVersion parameter come from?  Was it always there I guess?

why not use a query to get the version instead of relying on user input?  
(like https://github.com/keboola/db-extractor-mssql/blob/master/src/Keboola/DbExtractor/Extractor/MSSQL.php#L30)

Reply:
Yeah you're right. I'll query the DB. This parameter was always there, but it's not used now.

Reply:
@pivnicek wouldn't there be a problem with permissions for existing configurations? 
That the DB user in the configuration, don't have rights ti access the "SERVERPROPRTY" stuff?

Reply:
Shouldn't be, never happened on the ex.  I think this system info is available to all users.

Reply:
OK, so it is fine like this? There is now no test for the older version of the server though.